src/highdicom/sr/templates.py

```diff
@@ -1955,19 +3248,454 @@ def __init__(
+                )
+            elif item.value == codes.DCM.Fetus:
+                attributes = SubjectContextFetus.from_sequence(
+                    sequence=root_item.ContentSequence[index:next_index]
+                )
+            elif item.value == codes.DCM.Device:
+                attributes = SubjectContextDevice(
+                    sequence=root_item.ContentSequence[index:next_index]
+                )
+            else:
+                raise ValueError('Unexpected subject class "{item.meaning}".')
+            context = SubjectContext(
+                subject_class=item.value,
+                subject_class_specific_context=attributes
+            )
+            subject_contexts.append(context)
+        return subject_contexts
+
+    def get_planar_roi_measurement_groups(
+        self,
+        tracking_uid: Optional[str] = None,
+        finding_type: Optional[Union[CodedConcept, Code]] = None,
+        finding_site: Optional[Union[CodedConcept, Code]] = None,
+        reference_type: Optional[Union[CodedConcept, Code]] = None,
+        graphic_type: Optional[
+            Union[GraphicTypeValues, GraphicTypeValues3D]
+        ] = None,
+        referenced_sop_instance_uid: Optional[str] = None,
+        referenced_sop_class_uid: Optional[str] = None
+    ) -> List[PlanarROIMeasurementsAndQualitativeEvaluations]:
+        """Get imaging measurement groups of planar regions of interest.
+
+        Finds (and optionally filters) content items contained in the
+        CONTAINER content item "Measurement group" as specified by TID 1410
+        "Planar ROI Measurements and Qualitative Evaluations".
+
+        Parameters
+        ----------
+        tracking_uid: str, optional
+            Unique tracking identifier
+        finding_type: Union[highdicom.sr.CodedConcept, pydicom.sr.coding.Code], optional
+            Finding
+        finding_site: Union[highdicom.sr.CodedConcept, pydicom.sr.coding.Code], optional
+            Finding site
+        reference_type: Union[highdicom.sr.CodedConcept, pydicom.sr.coding.Code], optional
+            Type of referenced ROI
+        graphic_type: Union[highdicom.sr.GraphicTypeValues, highdicom.sr.GraphicTypeValues3D], optional
+            Graphic type of image region
+        referenced_sop_instance_uid: str, optional
+            SOP Instance UID of the referenced instance, which may be a
+            segmentation image, source image for the region or segmentation, or
+            RT struct, depending on `reference_type`
+        referenced_sop_class_uid: str, optional
+            SOP Class UID of the referenced instance, which may be a
+            segmentation image, source image for the region or segmentation, or
+            RT struct, depending on `reference_type`
+
+        Returns
+        -------
+        List[highdicom.sr.PlanarROIMeasurementsAndQualitativeEvaluations]
+            Sequence of content items for each matched measurement group
+
+        """  # noqa: E501
```

Comment:
Should we add checks that a user has provided a sensible combination of inputs? For example it makes no sense to ask for `reference_type` of referenced segment, and also provide a value for `graphic_type` right?

Reply:
I'm not sure it makes sense to just ignore the parameter if they are used in an "and" fashion, because it would need to match both parameters to be returned, which is impossible. Which one would be ignored? Presumably under the current behaviour it would just return an empty list every time this combination of parameters is passed regardless of the contents of the SR, which is ok but an error message telling users that they've asked for something impossible seems potentially helpful? I'm ok with leaving as it is